rmw_iceoryx_cpp/src/rmw_qos.cpp

```diff
@@ -26,13 +26,28 @@ rmw_qos_profile_check_compatible(
   char * reason,
   size_t reason_size)
 {
-  RCUTILS_CHECK_ARGUMENT_FOR_NULL(compatibility, RMW_RET_ERROR);
-  RCUTILS_CHECK_ARGUMENT_FOR_NULL(reason, RMW_RET_ERROR);
+  RCUTILS_CHECK_ARGUMENT_FOR_NULL(compatibility, RMW_RET_INVALID_ARGUMENT);
+  RCUTILS_CHECK_ARGUMENT_FOR_NULL(reason, RMW_RET_INVALID_ARGUMENT);
   (void)publisher_profile;
   (void)subscription_profile;
   (void)reason_size;
 
-  RMW_SET_ERROR_MSG("rmw_iceoryx_cpp does not support QoS profile check compatible");
-  return RMW_RET_UNSUPPORTED;
+  /// @todo Decide if implementing a matching function which takes subscriber and publisher
```

Comment:
I talked to @MatthiasKillat yesterday about whether it makes sense to implement something like `check_compatibility(PublisherOptions, SubscriberOptions)` in iceoryx. My conclusion after checking the criteria: It doesn't make sense. I'd suggest to move the comment block to the iceoryx documentation on iceoryx.io and remove them here.

@elBoberido What do you think?

Reply:
I thin having something like this in iceoryx make sense, at least as documentation. We should also have this in the introspection to show that something cannot be connected because of the options and not because of the service description. Having a method to call does not make that much sense to me but if ROS has all the settings available and can perform this check, iceoryx could provide this functionality

Reply:
Moved the comments to [this PR](https://github.com/eclipse-iceoryx/iceoryx/pull/1015/files).